Firmware/LowLevel/src/datatypes.h

```diff
@@ -135,22 +137,62 @@ struct ll_ui_event {
 #define LL_HIGH_LEVEL_CONFIG_BIT_DFPIS5V 1 << 0            // Enable full sound via mower_config env var "OM_DFP_IS_5V"
-#define LL_HIGH_LEVEL_CONFIG_BIT_EMERGENCY_INVERSE 1 << 1  // Sample, for possible future usage, i.e. for SA-Type emergency
+#define LL_HIGH_LEVEL_CONFIG_BIT_BACKGROUND_SOUNDS 1 << 1  // Enable background sounds
 
 typedef char iso639_1[2]; // Two char ISO 639-1 language code
 
+enum class HallMode : uint8_t {
+    off = 0,
+    emergency,  // lift & tilt
+    stop,
+    pause
+};
+
+// FIXME: Decide later which is more comfortable, activeLow = 0 | 1
+enum class HallLevel : uint8_t {
+    activeLow = 0,  // If Hall-Sensor (or button) is active/triggered we've this level on our GPIO
+    activeHigh
+};
+
+#pragma pack(push, 1)
+struct HallConfig {
+    HallMode mode : 4;
+    HallLevel level : 1;
+} __attribute__((packed));
+#pragma pack(pop)
+
+#define MAX_HALL_INPUTS 10  // How much Hall-inputs we support. 4 * OM + 6 * Stock-CoverUI + 0 spare (because not yet required to make it fixed)
+
+// LL/HL config packet, bi-directional, flexible-length, with defaults for YF-C500.
 #pragma pack(push, 1)
 struct ll_high_level_config {
-    uint8_t type;
-    uint8_t comms_version = LL_HIGH_LEVEL_CONFIG_MAX_COMMS_VERSION;  // Increasing comms packet-version number for packet compatibility (n > 0)
-    uint8_t config_bitmask = 0;                                      // See LL_HIGH_LEVEL_CONFIG_BIT_*
-    int8_t volume;                                                   // Volume (0-100%) feedback (if directly changed via CoverUI)
-    iso639_1 language;                                               // ISO 639-1 (2-char) language code (en, de, ...)
-    uint16_t spare1 = 0;                                             // Spare for future use
-    uint16_t spare2 = 0;                                             // Spare for future use
-    uint16_t crc;
+    // ATTENTION: This is a flexible length struct. It is allowed to grow independently to HL without loosing compatibility,
+    //    but never change or restructure already published member, except you really know their consequences.
+
+    // uint8_t type; Just for illustration. Get set in wire buffer with type PACKET_ID_LL_HIGH_LEVEL_CONFIG_REQ or PACKET_ID_LL_HIGH_LEVEL_CONFIG_RSP
+
+    uint8_t config_bitmask = 0;                    // See LL_HIGH_LEVEL_CONFIG_BIT_*
+    uint16_t rain_threshold = 700;                 // If (stock CoverUI) rain value < rain_threshold then it rains. Expected to differ between C500, SA and SC types
+    float v_charge_cutoff = V_CHARGE_MAX;          // Protective max. charging voltage before charging get switched off
+    float i_charge_cutoff = I_CHARGE_MAX;          // Protective max. charging current before charging get switched off
+    float v_battery_cutoff = 29.0f;                // Protective max. battery voltage before charging get switched off
```

Comment:
Since you started adding constants, this could need one as well 😉

Reply:
Yes good point :-)
This is in contrary to what I wrote before :-/

v_battery_cutoff was v_battery_max before, even it was not a max in my sense, but max as macro was redundant, so I took the static value.
With the integration of the new v_battery_empty/full values, _max couldn't stay as they are. So I rename that whole group as well as their macros... but didn't recognized that cutoff can become a macro now.
Well, changed now for the _cutoff value but not for rain_threshold... undecided if it make sense to give minor static values a macro if they're only used at one place :-/